rustls/src/client/hs.rs

```diff
@@ -1181,6 +1189,33 @@ impl State<ClientConnectionData> for ExpectServerHelloOrHelloRetryRequest {
     }
 }
 
+#[derive(Debug)]
+struct CertificateTypeExtensionParams {
+    server_negotiated: Option<CertificateType>,
+    client_expects: CertificateType,
+    extension_type: ExtensionType,
+}
+
+impl CertificateTypeExtensionParams {
+    fn validate_certificate_type_extension(&self) -> CertificateTypeExtensionNegotationResult {
+        let server_supports_raw_key = self.server_negotiated == Some(CertificateType::RawPublicKey);
+        let client_expects_raw_key = self.client_expects == CertificateType::RawPublicKey;
+        match (client_expects_raw_key, server_supports_raw_key) {
```

Comment:
It seems like you could `match` `server_negotiated` and `client_expects` directly instead of first defining a `bool`, which would make this logic easier to follow.

Reply:
Yup, much better.